### PR TITLE
Adding action to probe the releasability of this module

### PR DESCRIPTION
**Pull request by n3wscott**
This is a PoC of a new action we might want to help with releases.

### CODE REVIEW

File: .github/workflows/knative-releasability.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# repo by knobots: https://github.com/mattmoor/knobots and will be overwritten.
+
+name: 'Releasability'
+
+on:
+  schedule:
+  - cron: '0 1 * * *'
+
+jobs:
+  releasability:
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:25]: below is not indented enough 🤣

File: .github/workflows/knative-releasability.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    SLACK_ICON: http://github.com/knative.png?size=48
+    SLACK_USERNAME: github-actions
+    SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+    SLACK_CHANNEL: '@n3wscott'
+    MSG_MINIMAL: 'true'
+  
+  strategy:
+    matrix:
+      go-version: [1.15.x]
+      platform: [ubuntu-latest]
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:37]: I don't think this will be a matrix

File: .github/workflows/knative-releasability.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+jobs:
+  releasability:
+  name: Releasability
+  env:
+    SLACK_ICON: http://github.com/knative.png?size=48
+    SLACK_USERNAME: github-actions
+    SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+    SLACK_CHANNEL: '@n3wscott'
+    MSG_MINIMAL: 'true'
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:32]: I think it's clearer inline 🤔

File: .github/workflows/knative-releasability.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      run: |
+        buoy check go.mod --release $RELEASE --verbose
+
+    - name: Post success notice to Slack
+      uses: rtCamp/action-slack-notify@v2.1.0
+      env:
+        SLACK_COLOR: '#098e00'
+        SLACK_TITLE: Releasability for ${{ github.repository }} @ $RELEASE is GO!
+        SLACK_MESSAGE: |
+          For detailed logs: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:71]: I'd make this `if: ${{ success() }}` just for clarity

File: .github/workflows/knative-releasability.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      uses: actions/checkout@v2
+
+    - name: Install Dependencies
+      run: |
+        GO111MODULE=on go get tableflip.dev/buoy@master
+
+    - name: Next
+      run: |
+        export RELEASE=`buoy next go.mod`
+        echo "::set-env name=RELEASE::$RELEASE"
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:59]: This is like the one thing I'd actually put above 🤣

File: .github/workflows/knative-releasability.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+  runs-on: ${{ matrix.platform }}
+
+  steps:
+
+    - name: Set up Go ${{ matrix.go-version }}
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+      id: go
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:54]: I don't think we need `id: go` anywhere

File: .github/workflows/knative-releasability.yaml (hunk 1 above)
Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml:25]: let me go get my ruler

File: .github/workflows/knative-releasability.yaml (hunk 2 above)
Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml]: 👍

File: .github/workflows/knative-releasability.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15.x
+
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Install Dependencies
+        run: |
+          GO111MODULE=on go get tableflip.dev/buoy@master
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:52]: This needs to happen outside of the module or it will modify go.mod

Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml:52]: that might matter later, ok will fix

File: .github/workflows/knative-releasability.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+      - name: Check
+        run: |
+          buoy check go.mod --release $RELEASE --verbose
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:56]: After checking this, you need to `./hack/update-deps.sh --upgrade && ./hack/update-codegen.sh` and then check for diffs.

Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml:56]: ack, followup

**Review by mattmoor — COMMENTED** [review]
/lgtm

**Pull request merged**